### ISSUE TITLE
empty is nil in unmarshal

### ISSUE THREAD

**Issue by dezren39** [sblinch/kdl-go#4]
empty nodes are nil in unmarshal. this makes sense to me, but has caused me inconvenience and i wondered if it made sense to put an empty interface there. when using unmarshal, kdl behaved differently than the others when looking at koanf
![image](https://github.com/sblinch/kdl-go/assets/11225574/fadab377-57e2-465f-905c-96b6cae9bebc)
i believe there might be something in kdl specs that says the data model or something agrees with what you've done, i just, wanted to note this issue here and maybe i'm doing something wrong.

https://github.com/developing-today-forks/koanf/pull/1/files

**Comment by dezren39** [comment]
so with this
```
key "value"
map "skipped"
map key="skipped" key="value"
nested_map {
map key="value" 17 {
	list "item1" "item2" "item3"
	mixup ""=1 2 3 4
	first "first"=1 2 3 4
	child ""=1 2 3 4 { "" 5 ; "" 6 ; }
}
}
```

i get unmarshal errors from map being there twice at start, i think because the type changes. but i think it should just replace the type and the last one should win? any other option gets complicated, but i dont think it should throw because its valid kdl and there is a set order so replacing makes sense.

separate issue `{ "" 5 ; "" 6 ; }` is valid kdl but seems to cause an issue. i think semicolon isn't correctly being handled.
```
{
"" 5
"" 6 }
```
seems to work better but i think that maybe empty node name is behaving strangely. perhaps identifiers have been given the rules for bare identifiers? for instance, i don't think i can make a "1" node, yet that is all valid kdl so long as double quotes are used?

i'm also having issues with properly unmarshalling nested_map, it doesn't panic but it returns empty. this problem may be unrelated to kdl-go though, not sure yet. nobody likes empty string keys :)

i did try and read through https://github.com/sblinch/kdl-go/blob/main/docs/unmarshal.md but i didn't find anything that said these documents should unmarshal. i appreciate the detailed steps and everything you listed there. i've updated my plans in other areas to match how you've chosen ordering arguments, then properties, then children. i had thought most toggles were to extend what is permissible past spec. is there a toggle that enables overwriting values or something i'm missing that would be causing these kdl documents to error?

it makes sense to me to have flags to control behavior for unmarshalling into a map, sometimes i want to import a kdl doc and explode when any nodes and/or props overlap, sometimes i just want whatever i get at the end. there's a ton of complexity there but idk about all that maybe just a flag to invert default behavior to start. unmarshalling is complicated, obviously i'd prefer to use structs with hints but sometimes i can't. trying to avoid needing to parse into a kdl document for wider variety of cases. 😅

**Comment by sblinch** [comment]
Please provide a complete test case including the Go code and the KDL you're unmarshaling. The struct tags you use will completely change how a KDL snippet is interpreted so I can't even begin to diagnose without that information.

Please also open a separate issue for each bug you're reporting as it becomes confusing tracking multiple issues in a single thread.

**Comment by dezren39** [comment]
I'll make an isolated example when I get back to that machine, for now here is my marshal unmarshal commands

```
func (p *KDL) Unmarshal(b []byte) (map[string]interface{}, error) {
        var o map[string]interface{}
        err := kdl.Unmarshal(b, &o)
        return o, err
}
func (p *KDL) Marshal(o map[string]interface{}) ([]byte, error) {
        return kdl.Marshal(o)
}
```
Thank you!

**Comment by sblinch** [comment]
Fixed in 3d5e45f.
